### PR TITLE
[Task-6] : Redux

### PR DESCRIPTION
**Pull request by Mazorakopf**

### CODE REVIEW

File: src/pages/MovieSearch/action.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  release_date: model.release_date,
+  poster_path: model.poster_path,
+  overview: model.poster_path,
+  budget: model.budget,
+  revenue: model.revenue,
+  genres: Array.from([model.genre]),
+  runtime: Number(model.runtime),
+});
+
+const getMovies = (query) => async (dispatch) => {
```
Comment (Vladlan) [inline src/pages/MovieSearch/action.js]: Is in this file you have actions or actionCreators? If yes rename file than.

File: src/pages/MovieSearch/action.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+      return dispatch({
+        type: ActionTypes.RESET_MOVIE_MODAL,
+        payload: toModel(result),
+      });
+    });
+
+export const closeMovieModal = (dispatch) => (modalType) =>
+  dispatch({ type: ActionTypes.CLOSE_MOVIE_MODAL, modalType });
+
+export const onInputChanged = (dispatch, { selectedMovie }) => (event) =>
```
Comment (Vladlan) [inline src/pages/MovieSearch/action.js]: All names should reflect what they will do with store. onInputChanged saying nothing about it

File: src/pages/MovieSearch/action.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+export const fetchAllMovies = (dispatch) => () => dispatch(getMovies());
+
+export const openMovieDetails = (dispatch) => (selectedMovie) =>
+  dispatch({ type: ActionTypes.OPEN_MOVIE_DETAILS, payload: selectedMovie });
+
+export const openMovieModal = (dispatch) => (modalType, selectedMovie) =>
+  dispatch({
+    type: ActionTypes.OPEN_MOVIE_MODAL,
+    modalType,
```
Comment (Vladlan) [inline src/pages/MovieSearch/action.js]: dispatch should always receive an object {type: ..., payload:...}

File: src/pages/MovieSearch/index.js [hunk 4]
```diff
@@ -1,43 +1,59 @@
...
+const mapDispatchToProps = (dispatch) => ({
+  fetchAllMovies: Actions.fetchAllMovies(dispatch),
+  openMovieDetails: Actions.openMovieDetails(dispatch),
+  openMovieModal: Actions.openMovieModal(dispatch),
+  openMovieSearchForm: Actions.openMovieSearchForm(dispatch),
+  sortMovies: Actions.sortMovies(dispatch),
+  filterMoviesByGenre: Actions.filterMoviesByGenre(dispatch),
+});
+
+export default connect(mapStateToProps, mapDispatchToProps)(MovieSearch);
```
Comment (Vladlan) [inline src/pages/MovieSearch/index.js]: The main idea or redux is to reduce cases when you will pass props from one component to another. Use connect for other components too and remove props waterfall.

File: src/components/MovieList/MovieNavBar/index.js [hunk 5]
```diff
@@ -1,19 +1,27 @@
...
+      <select name="sortOptions" onChange={sortMovies}>
+        <option></option>
+        <option value="release_date">release data</option>
+        <option value="rating">rating</option>
       </select>
     </li>
   </ul>
 );
+
+MovieNavBar.propTypes = {
```
Comment (Vladlan) [inline src/components/MovieList/MovieNavBar/index.js]: App is not working because of this file

File: src/pages/MovieSearch/action.js [hunk 6]
```diff
@@ -0,0 +1,139 @@
+import { MovieModalType } from '../../components/MovieModal/util';
+
+const baseUrl = 'http://localhost:4000';
```
Comment (Vladlan) [inline src/pages/MovieSearch/action.js]: Add backend for you repo for this

File: src/pages/MovieSearch/action.js [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+
+const getMovies = (query) => async (dispatch) => {
+  let url = `${baseUrl}/movies`;
+  if (query?.sortBy) {
+    url += `?sortBy=${query.sortBy}&sortOrder=desc`;
+  }
+  if (query?.filter) {
+    url += `?searchBy=genres&filter=${query.filter}`;
+  }
+  const response = await fetch(url, { method: 'GET' });
```
Comment (Vladlan) [inline src/pages/MovieSearch/action.js]: How about to add error handling?

**Review by Vladlan — CHANGES_REQUESTED** [review]

File: src/pages/MovieSearch/action.js (hunk 2 above)
Comment (Mazorakopf) [reply src/pages/MovieSearch/action.js]: renamed to `changeSelectedMovie`

File: src/components/MovieList/MovieNavBar/index.js (hunk 5 above)
Comment (Mazorakopf) [reply src/components/MovieList/MovieNavBar/index.js]: fixed

File: src/pages/MovieSearch/action.js (hunk 3 above)
Comment (Mazorakopf) [reply src/pages/MovieSearch/action.js]: fixed

File: src/pages/MovieSearch/action.js (hunk 6 above)
Comment (Mazorakopf) [reply src/pages/MovieSearch/action.js]: moved backend calls to api package

File: src/pages/MovieSearch/action.js (hunk 7 above)
Comment (Mazorakopf) [reply src/pages/MovieSearch/action.js]: added

**Pull request merged**